x-pack/docs/en/security/auditing/event-types.asciidoc

```diff
@@ -2,63 +2,146 @@
+| `{"type":"audit", "timestamp":"2020-12-31T00:12:09,349+0200", "node.id":"0RMNyghkQYCc_gVd1G6tZQ", "event.type":"security_config_change", "event.action":"delete_role_mapping", "request.id":"Stim-DuoSTCWom0S_xhf8g", "delete":{"role_mapping":{"name":"mapping1"}}}`
+
+| `delete_user`
+| Logged when the <<security-api-delete-user,delete user API>> is invoked to
+delete a specific native user.
+| `{"type":"audit", "timestamp":"2020-12-30T22:19:41,345+0200", "node.id":"0RMNyghkQYCc_gVd1G6tZQ", "event.type":"security_config_change", "event.action":"delete_user", "request.id":"au5a1Cc3RrebDMitMGGNCw", "delete":{"user":{"name":"jacknich"}}}`
+
+| `invalidate_apikeys`
+| Logged when the <<security-api-invalidate-api-key,invalidate API key API>> is
+invoked to invalidate one or more API keys.
+| `{"type":"audit", "timestamp":"2020-12-31T00:36:30,247+0200", "node.id":"9clhpgjJRR-iKzOw20xBNQ", "event.type":"security_config_change", "event.action":"invalidate_apikeys", "request.id":"7lyIQU9QTFqSrTxD0CqnTQ", "invalidate":{"apikeys":{"owned_by_authenticated_user":false,"user":{"name":"myuser","realm":"native1"}}}}`
+
+| `put_privileges`
+| Logged when the <<security-api-put-privileges,put privileges API>> is invoked
+to add or update one or more application privileges.
+| `{"type":"audit", "timestamp":"2020-12-31T00:39:07,779+0200", "node.id":"9clhpgjJRR-iKzOw20xBNQ", "event.type":"security_config_change", "event.action":"put_privileges", "request.id":"1X2VVtNgRYO7FmE0nR_BGA", "put":{"privileges":[{"application":"myapp","name":"read","actions":["data:read/*","action:login"],"metadata":{"description":"Read access to myapp"}}]}}`
+
+| `put_role`
+| Logged when the <<security-api-put-role,put role API>> is invoked to create or
+update a role.
+| `{"type":"audit", "timestamp":"2020-12-30T22:27:01,978+0200", "node.id":"0RMNyghkQYCc_gVd1G6tZQ", "event.type":"security_config_change", "event.action":"put_role", "request.id":"tDYQhv5CRMWM4Sc5Zkk2cQ", "put":{"role":{"name":"test_role","role_descriptor":{"cluster":["all"],"indices":[{"names":["apm*"],"privileges":["all"],"field_security":{"grant":["granted"]},"query":"{\"term\": {\"service.name\": \"bar\"}}"},{"names":["apm-all*"],"privileges":["all"],"query":"{\"term\": {\"service.name\": \"bar2\"}}"}],"applications":[],"run_as":[]}}}}`
+
+| `put_role_mapping`
+| Logged when the <<security-api-put-role-mapping,put role mapping API>> is
+invoked to create or update a role mapping.
+| `{"type":"audit", "timestamp":"2020-12-31T00:11:13,932+0200", "node.id":"0RMNyghkQYCc_gVd1G6tZQ", "event.type":"security_config_change", "event.action":"put_role_mapping", "request.id":"kg4h1l_kTDegnLC-0A-XxA", "put":{"role_mapping":{"name":"mapping1","roles":["user"],"rules":{"field":{"username":"*"}},"enabled":true,"metadata":{"version":1}}}}`
+
+| `put_user`
+| Logged when the <<security-api-put-user,put user API>> is invoked to create or
+update a native or built-in user. Note that user updates can also change the
```

Comment:
```suggestion
update a native user. Note that user updates can also change the
```

You cannot call the put user API on a reserved user.

Reply:
Ha! yeah, that's right. I think I remember looking specifically about that, but I must've not looked in the transport action for the validation.
Though, technically, the `put_user` event will be emitted for reserved users too, because auditing happens before validation.
But I'll go with your suggestion.